homeassistant/components/backup/manager.py

```diff
@@ -566,6 +566,33 @@ async def async_create_backup(
         include_homeassistant: bool,
         name: str | None,
         password: str | None,
+    ) -> NewBackup:
+        """Create a backup."""
+        new_backup = await self.async_initiate_backup(
+            agent_ids=agent_ids,
+            include_addons=include_addons,
+            include_all_addons=include_all_addons,
+            include_database=include_database,
+            include_folders=include_folders,
+            include_homeassistant=include_homeassistant,
+            name=name,
+            password=password,
+        )
+        assert self._backup_finish_task
+        await self._backup_finish_task
+        return new_backup
```

Comment:
Maybe this should no longer return the `new_backup`, but instead the `AgentBackup` which we get when awaiting the task?

Reply:
I've been thinking about that too. NewBackup is a detail I think. I can look at this later when I'm finished with the error handling.